### PR TITLE
Fix astropy.visualization test warnings

### PR DESCRIPTION
**Pull request by astrofrog**
This reduces the number of warnings that occur during the testing in astropy.visualization from 590 to 2 :tada:

### CODE REVIEW

File: astropy/visualization/tests/test_time.py [hunk 1]
```diff
@@ -124,7 +135,11 @@ def test_formats(format, expected):
     with time_support(format=format, simplify=False):
         fig = plt.figure()
         ax = fig.add_subplot(1, 1, 1)
-        ax.set_xlim(Time('2014-03-22T12:30:30.9'), Time('2077-03-22T12:30:32.1'))
+        # Getting unix time and plot_date requires going through a scale for
+        # which ERFA emits a warning about the date being dubious
+        with pytest.warns(ErfaWarning) if format in ['unix', 'plot_date'] else nullcontext():
```
Comment (pllim) [inline astropy/visualization/tests/test_time.py]: Oh, nice! TIL.

Comment (astrofrog) [reply astropy/visualization/tests/test_time.py]: me too haha :)

File: astropy/wcs/wcsapi/fitswcs.py [hunk 2]
```diff
@@ -395,6 +395,8 @@ def _get_components_and_classes(self):
                             location = EarthLocation(*self.wcs.obsgeo[:3], unit=u.m)
                     elif trefpos == 'geocenter':
                         location = EarthLocation(0, 0, 0, unit=u.m)
+                    elif trefpos == '':
```
Comment (pllim) [inline astropy/wcs/wcsapi/fitswcs.py]: I guess it is a real API change outside of testing, so I'll wait for @Cadair or @nden to have a look first.

**Comment by pllim** [comment]
xref #7928

File: astropy/wcs/wcsapi/fitswcs.py (hunk 2 above)
Comment (Cadair) [reply astropy/wcs/wcsapi/fitswcs.py]: This is above my pay grade :stuck_out_tongue:

File: astropy/visualization/wcsaxes/tests/test_wcsapi.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ def wcs_4d():
     LONPOLE =                180.0 / [deg] Native longitude of celestial pole
     LATPOLE =                  0.0 / [deg] Native latitude of celestial pole
     """)
-    return WCS(header=header)
+    return WCS(header=fits.Header.fromstring(header, sep='\n'))
```
Comment (Cadair) [inline astropy/visualization/wcsaxes/tests/test_wcsapi.py]: Well! that's a better fix than what I threw together in SunPy.

File: astropy/wcs/wcsapi/fitswcs.py (hunk 2 above)
Comment (astrofrog) [reply astropy/wcs/wcsapi/fitswcs.py]: It's an affects-dev change, just fixes a bug that was introduced in a recent PR.

**Pull request merged**